### PR TITLE
[fix] Fix locked versions feeding back into new lock state

### PR DESCRIPTION
**Pull request by dansanduleac**
## Before this PR

Addresses another root cause of https://github.com/palantir/gradle-consistent-versions/issues/117.
Specifically, when deciding the "current" lock state as part of the `verifyLocks` task, current constraints coming from `vesions.lock` are taken into account when doing that resolution.

As an example, a version of `1.2.0` from `versions.lock` will artifically raise the version in the "current lock state", even if the constraints have since changed such that they only require `1.1.0`. In that case, `verifyLocks` would think everything is up to date, even if `--write-locks` would actually produce a different lock file (with the correct new version of `1.1.0`).

## After this PR
==COMMIT_MSG==
Fix locked versions feeding back into new lock state. `verifyLocks` determines current lock state completely independently of the current versions in `versions.lock`.
==COMMIT_MSG==

We also simplified how we configure the recursive dependency copying for unifiedClasspath - instead of relying on the laziness of `withDependencies`, which was hard to follow, we just call the `recursivelyCopyProjectDependencies` method at the appropriate time in each case (writing locks vs not).

### CODE REVIEW

File: src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java [hunk 1]
```diff
@@ -298,7 +302,7 @@ private void recursivelyCopyProjectDependencies(
             Project currentProject, DependencySet dependencySet, Map<Configuration, String> copiedConfigurationsCache) {
         dependencySet
                 .matching(dependency -> ProjectDependency.class.isAssignableFrom(dependency.getClass()))
-                .configureEach(dependency -> {
+                .all(dependency -> {
```
Comment (iamdanfox) [inline src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java:306]: configureEach is lazy and all is not lazy, and apparently this change wasn't super essential but nice to be defensive anyway.

**Review by iamdanfox — APPROVED** [review]

**Comment by dansanduleac** [comment]
👍
